R/wwinference.R

```diff
@@ -27,6 +27,10 @@
 #' user based on the date they are producing a forecast
 #' @param mcmc_options The MCMC parameters as defined using
 #' `get_mcmc_options()`.
+#' @param spec_inits Boolean indicating whether or not to specify the
+#' initialization of the sampler, default is `TRUE`, meaning that the
+#' initialization lists will be passed to the `cmdstanr::sample()`
+#' function
```

Comment:
Passing `init = NULL` to the `sample()` method of a `CmdStanModel` object leads to random initis (see [docs](mdstanr/reference/model-method-sample.html).

So instead of having the if/else on `spec_inits` in your fitting code, I'd do something like:

```R
init_lists <- NULL
if(spec_inits) {
  init_lists <- c()
  for (i in 1:mcmc_options$n_chains) {
    init_lists[[i]] <- get_inits(stan_data, params)
  }
}
```

I'd also consider naming this flag something like `generate_initial_values`, since `spec_inits` slightly clashes with the `model_spec` argument.